sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -408,8 +408,8 @@ private[hive] class HiveClientImpl(
         lastAccessTime = h.getLastAccessTime.toLong * 1000,
         storage = CatalogStorageFormat(
           locationUri = shim.getDataLocation(h),
-          inputFormat = Option(h.getInputFormatClass).map(_.getName),
-          outputFormat = Option(h.getOutputFormatClass).map(_.getName),
+          inputFormat = Option(h.getTTable.getSd.getInputFormat),
+          outputFormat = Option(h.getTTable.getSd.getOutputFormat),
```

Comment:
After more readings, `getTTable.getSd.getInputFormat` and `getTTable.getSd.getOutputFormat` will be null for non-native Hive tables, e.g., JDBC tables, HBase tables and Cassandra tables. See the link for more details: https://cwiki.apache.org/confluence/display/Hive/StorageHandlers

So far, this is OK. I am just afraid we might expand the usage of `getTableOption` in the future. Maybe at least document the restrictions?